lib/boxr/chunked_uploads.rb

```diff
@@ -9,11 +9,13 @@ def chunked_upload_create_session_new_file(path_to_file, parent, name: nil)
       end
     end
 
-    def chunked_upload_create_session_new_file_from_io(io, parent, name)
+    def chunked_upload_create_session_new_file_from_io(io, parent, name, io_size: nil)
+      io_size ||= io.size
```

Comment:
One thing to keep in mind is that `#size` is **not** part of `IO`'s public interface.